atd-vze/src/views/Locations/Location.js

```diff
@@ -41,12 +41,57 @@ function Location(props) {
   if (error) return `Error! ${error.message}`;
 
   const getAggregatePersonsSum = (data, field) => {
+    console.log(
+      data.atd_txdot_locations[0],
+      data.atd_txdot_locations[0].crashes_by_manner_collision.map(
+        a => a.collsn_desc
+      )
+    );
     return (
       data.atd_txdot_primaryperson_aggregate.aggregate.sum[field] +
       data.atd_txdot_person_aggregate.aggregate.sum[field]
     );
   };
 
+  const formatLabel = str => {
+    var maxwidth = parseInt(str.length / 1.75);
+    var sections = [];
+    var words = str.split(" ");
+    var temp = "";
+
+    words.forEach(function(item, index) {
+      if (temp.length > 0) {
+        var concat = temp + " " + item;
+
+        if (concat.length > maxwidth) {
+          sections.push(temp);
+          temp = "";
+        } else {
+          if (index === words.length - 1) {
+            sections.push(concat);
+            return;
+          } else {
+            temp = concat;
+            return;
+          }
+        }
+      }
+
+      if (index === words.length - 1) {
+        sections.push(item);
+        return;
+      }
+
+      if (item.length < maxwidth) {
+        temp = item;
+      } else {
+        sections.push(item);
+      }
+    });
+
+    return sections;
+  };
```

Comment:
@tcwcodes, this is a pretty complex function that took me a full 30 minutes to dig through and understand. If I may make a suggestion for a more simple method, how would you feel about this... (it assumes we only need to split the string in half because I noticed the ones that wrapped to 3 lines looked a little crammed)

```js
  const formatLabel = str => {
    let sections = [];

    // Get the approx midpoint of string
    let splitPoint = Math.floor(str.length / 2);

    // If the midpoint is not a space,
    // find the closest " " to the left of the midpoint and split there instead
    if (str.charAt(splitPoint) !== " ") {
      splitPoint = str.substring(0, splitPoint).lastIndexOf(" ");
    }

    sections.push(str.substring(0, splitPoint));
    sections.push(str.substring(splitPoint));

    return sections;
  };
```

Result:

![Screen Shot 2019-10-09 at 4 10 05 PM](https://user-images.githubusercontent.com/5697474/66520721-55bae980-eaaf-11e9-8150-ea24c0c1df71.png)

Reply:
This is much better/cleaner! I went ahead with this function.